ext/node/ops/v8.rs

```diff
@@ -30,3 +39,343 @@ pub fn op_v8_get_heap_statistics(
+  // alive while we hold a reference to it.
+  _backing_store: v8::SharedRef<v8::BackingStore>,
+}
+
+pub struct Deserializer<'a> {
+  buf: DeserBuffer,
+  inner: v8::ValueDeserializer<'a>,
+}
+
+impl<'a> deno_core::GarbageCollected for Deserializer<'a> {}
+
+pub struct DeserializerDelegate {
+  obj: v8::Global<v8::Object>,
+}
+
+impl GarbageCollected for DeserializerDelegate {
+  fn trace(&self, _visitor: &v8::cppgc::Visitor) {}
+}
+
+impl v8::ValueDeserializerImpl for DeserializerDelegate {
+  fn read_host_object<'s>(
+    &self,
+    scope: &mut v8::HandleScope<'s>,
+    _value_deserializer: &dyn v8::ValueDeserializerHelper,
+  ) -> Option<v8::Local<'s, v8::Object>> {
+    let obj = v8::Local::new(scope, &self.obj);
+    let key = FastString::from_static("_readHostObject")
+      .v8_string(scope)
+      .into();
+    let scope = &mut v8::AllowJavascriptExecutionScope::new(scope);
+    if let Some(v) = obj.get(scope, key) {
+      if let Ok(v) = v.try_cast::<v8::Function>() {
+        let result = v.call(scope, obj.into(), &[])?;
+        match result.try_cast() {
+          Ok(res) => return Some(res),
+          Err(_) => {
+            let msg =
+              FastString::from_static("readHostObject must return an object")
+                .v8_string(scope);
+            let error = v8::Exception::type_error(scope, msg);
+            scope.throw_exception(error);
+            return None;
+          }
+        }
+      }
+    }
+    None
+  }
+}
+
+#[op2]
+#[cppgc]
+pub fn op_v8_new_deserializer(
+  scope: &mut v8::HandleScope,
+  obj: v8::Local<v8::Object>,
+  buffer: v8::Local<v8::ArrayBufferView>,
+) -> Result<Deserializer<'static>, AnyError> {
+  let offset = buffer.byte_offset();
+  let len = buffer.byte_length();
+  let backing_store = buffer.get_backing_store().ok_or_else(|| {
+    generic_error("deserialization buffer has no backing store")
+  })?;
+  let (buf_slice, buf_ptr) = if let Some(data) = backing_store.data() {
+    // SAFETY: the offset is valid for the underlying buffer because we're getting it directly from v8
+    let data_ptr = unsafe { data.as_ptr().cast::<u8>().add(offset) };
+    (
+      // SAFETY: the len is valid, from v8, and the data_ptr is valid (as above)
+      unsafe { std::slice::from_raw_parts(data_ptr.cast_const().cast(), len) },
+      Some(data.cast()),
+    )
+  } else {
+    (&[] as &[u8], None::<NonNull<u8>>)
+  };
+  let obj = v8::Global::new(scope, obj);
+  let inner = v8::ValueDeserializer::new(
+    scope,
+    Box::new(DeserializerDelegate { obj }),
+    buf_slice,
+  );
+  Ok(Deserializer {
+    inner,
+    buf: DeserBuffer {
+      _backing_store: backing_store,
+      ptr: buf_ptr,
+    },
+  })
+}
+
+#[op2(fast)]
+pub fn op_v8_transfer_array_buffer_de(
+  #[cppgc] deser: &Deserializer,
+  #[smi] id: u32,
+  array_buffer: v8::Local<v8::ArrayBuffer>,
+) {
+  // TODO(nathanwhit): also need binding for TransferSharedArrayBuffer, then call that if
+  // array_buffer is shared
```

Comment:
Is there an issue in `rusty_v8` repo for that?

Reply:
Opened https://github.com/denoland/rusty_v8/issues/1615